app/service/collection.py

```diff
@@ -136,22 +137,32 @@ def update(
     :return Optional[CollectionDTO]: The updated collection or None if not updated.
     """
 
+    if db is None:
+        db = db_session.get_db()
+
     # TODO: implement changing of a collection's organisation
     # org_id = organisation.get_id_from_name(db, collection.organisation)
 
     validate_import_id(import_id)
 
-    collection_repo.update(db, import_id, collection)
-    db.commit()
+    transaction = db.begin_nested()
```

Comment:
@katybaulch  this will be the pattern we are to expect in delete/update functions - are you ok with this?